include/mxnet/lib_api.h

```diff
@@ -229,20 +241,53 @@ enum MXReturnValue {
   MX_SUCCESS = 1,
 };
 
+struct MXSparse {
+  // Pointer to data.
+  void *data{nullptr};
+  // length of (non-zero) data.
+  int64_t data_len;
+
+  // To store aux data for sparse.
+  // For CSR, indices stores the col index of non-zero values.
+  // For row sparse, indices store row index of rows which have non-zero values.
+  int64_t* indices;
+  int64_t indices_len;
+
+  // For CSR, indptr gives the start and end index of data for each row.
+  // For row sparse, indptr is empty. 
+  int64_t* indptr;
+  int64_t indptr_len;
+
+  void set(void *data_ptr, const int64_t* dims, int ndims, void *idx,
+          int64_t num_idx, void *idx_ptr = nullptr, int64_t num_idx_ptr = 0) {
+    data = data_ptr;
+    data_len = num_idx;
+
+    indices = (int64_t*)idx;
+    indices_len = num_idx;
+
+    if(idx_ptr) {
+      indptr = (int64_t*)idx_ptr;
+      indptr_len = num_idx_ptr;
+    }
+  }
```

Comment:
Can we move this into the MXTensor struct and call it `setSparse` or create another `setTensor` that also take these arguments too?

Reply:
Do you mean move this function to MXTensor? Is it better to keep it in MXSparse structure?